typeshed_client/finder.py

```diff
@@ -175,9 +175,11 @@ def _get_all_stub_files_from_directory(
     directory: _DirEntry, root_directory: Path, seen: Set[str]
 ) -> Generator[Tuple[str, Path], None, Set[str]]:
     new_seen = set(seen)
-    to_do: Set[os.PathLike[str]] = {directory}
+    # Use a dict for to_do to ensure modules are yielded in a deterministic order
+    # The keys in the dict are irrelevant; this is just a standin for an ordered set
+    to_do: Dict[os.PathLike[str], None] = {directory: None}
```

Comment:
I wonder if we could just use a list. Reading the code I'm not sure there's a way for the same file to appear multiple times.

Reply:
I assumed that there was a way for duplicates to creep in due to the use of a set currently, but I agree that a list would certainly be cleaner :)

Reply:
Updated the PR to use a list